projects/conding-challenge-i/src/App.tsx

```diff
@@ -0,0 +1,94 @@
+import { useState } from 'react'
+import './App.css'
+
+import CardName from './components/CardName'
+
+const randomInt = (min:number, max:number) => Math.floor(Math.random() * (max - min + 1)) + min;
+
+function findClosest(arr:number[],  target:number)
+{
+  let n = arr.length;
+
+  if (target <= arr[0])
+    return 0;
+
+  if (target >= arr[n-1])
+    return n-1;
+
+  let i = 0, j = n, mid= 0;
+
+  while(i<j)
+  {
+
+    mid = Math.floor((i+j) / 2);
+
+    if (arr[mid] == target)
+      return mid;
+
+    if (target < arr[mid]){
+      if (mid>0 && target>arr[mid-1])
+        return getClosest(mid - 1,
+                          mid, target, arr);
+      j = mid
+    }
+    else
+    {
+      if (mid < n - 1 && target < arr[mid+1])
+        return getClosest(mid, mid+1, target, arr);
+      i = mid + 1;
+    }
+  }
+
+  return mid;
+}
+
+function getClosest(curr_element:number,
+                    next_element:number,
+                    target:number, arr:number[]){
+  let val1 = arr[curr_element]
+  let val2 = arr[next_element]
+
+  if (Math.abs(val1-target) > Math.abs(val2 - target))
+  {
+    return next_element 
+  }
+  else{
+    return curr_element 
+  }
+}
+
+
+function App() {
+  const minRange = 1
+  const maxRange = 10
```

Comment:
ProTip: los litereales, o constantes que sirven unicamente como parametros para el calculo de otras funciones, se declaran en mayuscula, y fuera de la funcion del componente, ya que, al no cambiar su valor no es necesario que se regeneren en cada re-renderizado del componente
```suggestion
// Poner esto fuera de funcion App()
  const MINIMUN_RANGE = 1
  const MAXIMUN_RANGE = 10
```